src/test/regress/expected/node_conninfo_reload.out

```diff
@@ -297,8 +299,10 @@ select count(*) from test;
 
 COMMIT;
 -- Should fail now, when transaction is finished
+SET client_min_messages TO ERROR;
```

Comment:
There is a slight change in the failure handling, which I consider as a positive improvement, but good to hear your thoughts as well. All the test output changes are tied to the following. 

Previously, OpenNewConnections was calling into `ConnectionStateMachine` per new session.

Now, we let all necessary connections to be established, and then call into `ConnectionStateMachine` per session afterwards.

The implication of the above is that on `MULTI_CONNECTION_FAILED` we have `if (workerPool->failedConnectionCount >= totalConnectionCount)`, which behaves differently if there are multiple connections involved.

In the past, even if we had multiple connections, the **first** connection was triggering the error. Now, the **last** connection triggers the error. 

In the tests, there is no behavioral change, as all connections are failing. This only causes more WARNING. In the production workloads, maybe this can be useful. Say, the first connection is actually lost for some reason, but than the second (or any connection) succeeds. In that case, the execution might actually continue with the successful connections. Even a single successful connection is enough for the executor to finish.